File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
    var answer = [];
    
    function hanoi(n, from, to, via) {
        if (n === 1) {
            answer.push([from, to]);
        } else {
            hanoi(n - 1, from, via, to);
            hanoi(1, from, to, via);
            hanoi(n - 1, via, to, from);
        }
    }
    
    hanoi(n, 1, 3, 2);
    
    return answer;
}

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    answer = []
    
    def hanoi(n, off, to, via):
        if n == 1:
            answer.append([off, to])
        else:
            hanoi(n - 1, off, via, to)
            hanoi(1, off, to, via)
            hanoi(n - 1, via, to, off)
    
    hanoi(n, 1, 3, 2)
    
    return answer